chain/reorg_test.go

```diff
@@ -1,71 +1,141 @@
+	old, new, common := getPrefixOldNewCommon(ctx, t, chainStore, blockSource, dstP)
+	assert.False(t, chain.IsReorg(old, new, common))
+}
+
+func TestIsReorgSubset(t *testing.T) {
+	tf.UnitTest(t)
+	dstP := initDSTParams()
+	ctx, blockSource, chainStore := setupGetAncestorTests(t, dstP)
+	old, new, common := getSubsetOldNewCommon(ctx, t, chainStore, blockSource, dstP)
+	assert.False(t, chain.IsReorg(old, new, common))
+}
+
+func TestReorgDiffFork(t *testing.T) {
+	tf.UnitTest(t)
+	dstP := initDSTParams()
+	ctx, blockSource, chainStore := setupGetAncestorTests(t, dstP)
+	old, new, common := getForkOldNewCommon(ctx, t, chainStore, blockSource, dstP)
+
+	dropped, added, err := chain.ReorgDiff(old, new, common)
+	assert.NoError(t, err)
+	assert.Equal(t, uint64(11), dropped)
+	assert.Equal(t, uint64(14), added)
+}
+
+func TestReorgDiffSubset(t *testing.T) {
+	tf.UnitTest(t)
+	dstP := initDSTParams()
+	ctx, blockSource, chainStore := setupGetAncestorTests(t, dstP)
+	old, new, common := getSubsetOldNewCommon(ctx, t, chainStore, blockSource, dstP)
+
+	dropped, added, err := chain.ReorgDiff(old, new, common)
+	assert.NoError(t, err)
+	assert.Equal(t, uint64(1), dropped)
+	assert.Equal(t, uint64(1), added)
+}
+
+// getForkOldNewCommon
+func getForkOldNewCommon(ctx context.Context, t *testing.T, chainStore chain.Store, blockSource *th.TestFetcher, dstP *DefaultSyncerTestParams) (types.TipSet, types.TipSet, types.TipSet) {
+	// Add 10 tipsets to the head of the chainStore.
+	requireGrowChain(ctx, t, blockSource, chainStore, 10, dstP)
+	commonAncestor := requireHeadTipset(t, chainStore)
+
+	// make the first fork tipset
+	signer, ki := types.NewMockSignersAndKeyInfo(1)
+	mockSignerPubKey := ki[0].PublicKey()
+	fakeChildParams := th.FakeChildParams{
+		Parent:      commonAncestor,
+		GenesisCid:  dstP.genCid,
+		Signer:      signer,
+		MinerPubKey: mockSignerPubKey,
+		StateRoot:   dstP.genStateRoot,
+		Nonce:       uint64(4),
+	}
+
+	firstForkBlock := th.RequireMkFakeChild(t, fakeChildParams)
+	requirePutBlocks(t, blockSource, firstForkBlock)
+	firstForkTS := th.RequireNewTipSet(t, firstForkBlock)
+	firstForkTsas := &chain.TipSetAndState{
+		TipSet:          firstForkTS,
+		TipSetStateRoot: dstP.genStateRoot,
+	}
+	th.RequirePutTsas(ctx, t, chainStore, firstForkTsas)
+	err := chainStore.SetHead(ctx, firstForkTS)
+	require.NoError(t, err)
+
+	// grow the fork by 10 blocks (11 total)
+	requireGrowChain(ctx, t, blockSource, chainStore, 10, dstP)
+	oldHead := requireHeadTipset(t, chainStore)
+
+	// go back and complete the original chain
+	err = chainStore.SetHead(ctx, commonAncestor)
+	require.NoError(t, err)
+	requireGrowChain(ctx, t, blockSource, chainStore, 14, dstP)
+	newHead := requireHeadTipset(t, chainStore)
+
+	return oldHead, newHead, commonAncestor
+}
+
+func getPrefixOldNewCommon(ctx context.Context, t *testing.T, chainStore chain.Store, blockSource *th.TestFetcher, dstP *DefaultSyncerTestParams) (types.TipSet, types.TipSet, types.TipSet) {
```

Comment:
Ditto the above comment about method description and numbers as parameters here.